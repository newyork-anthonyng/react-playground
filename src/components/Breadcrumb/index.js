import React, { Component, Children } from "react";
import T from "prop-types";
import glamorous from "glamorous";

const List = glamorous.ol({
  display: "flex",
  listStyle: "none",
  padding: 0
});

const Crumb = glamorous.li({
  "&::after": {
    content: `''`,
    borderRight: "2px solid black",
    display: "inline-block",
    height: "16px",
    marginRight: "10px",
    paddingRight: "5px",
    transform: "rotate(13deg)",
    width: "3px"
  }
});

class Breadcrumb extends Component {
  constructor() {
    super();

    this.renderList = this.renderList.bind(this);
  }

  renderList() {
    const { children } = this.props;
    const childrenCount = Children.count(children);

    return Children.map(children, (link, i) => {
      if (i === childrenCount - 1) {
        return (
          <li>
            {React.cloneElement(link, { "aria-current": "page" })}
          </li>
        );
      }

      return (
        <Crumb>
          {link}
        </Crumb>
      );
    });
  }

  render() {
    const { children } = this.props;

    return (
      <nav aria-label="Breadcrumb">
        <List>
          {this.renderList()}
        </List>
      </nav>
    );
  }
}

Breadcrumb.propTypes = {
  children: T.node
};

export default Breadcrumb;
