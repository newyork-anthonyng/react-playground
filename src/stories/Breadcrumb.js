import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, number, boolean, text } from '@storybook/addon-knobs';

import Breadcrumb from '../components/Breadcrumb';

storiesOf('Breadcrumb ', module)
  .addDecorator(withKnobs)
  .add('With knobs', () => {
    return (
      <Breadcrumb>
        <a href="#1">First</a>
        <a href="#2">Second</a>
        <a href="#3">Third</a>
        <a href="#4">Fourth</a>
      </Breadcrumb>
    );
  })
