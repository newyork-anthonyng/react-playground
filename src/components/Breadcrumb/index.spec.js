import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Breadcrumb from './';

it('should render correctly', () => {
  const wrapper = shallow(
    <Breadcrumb>
      <a href="#one">One</a>
      <a href="#two">Two</a>
      <a href="#three">Three</a>
    </Breadcrumb>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render the last item with correct aria label', () => {
  const wrapper = shallow(
    <Breadcrumb>
      <a href="#one">One</a>
      <a href="#two">Two</a>
      <a href="#three">Three</a>
    </Breadcrumb>
  );
  const lastLink = wrapper.find('a').last();

  expect(lastLink.props()['aria-current']).toEqual('page');
});
