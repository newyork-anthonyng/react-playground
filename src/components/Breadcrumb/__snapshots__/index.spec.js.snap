// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<nav
  aria-label="Breadcrumb"
>
  <glamorous(ol)>
    <glamorous(li)>
      <a
        href="#one"
      >
        One
      </a>
    </glamorous(li)>
    <glamorous(li)>
      <a
        href="#two"
      >
        Two
      </a>
    </glamorous(li)>
    <li>
      <a
        aria-current="page"
        href="#three"
      >
        Three
      </a>
    </li>
  </glamorous(ol)>
</nav>
`;
